providers/implementations/keymgmt/ml_dsa_kmgmt.c

```diff
@@ -42,6 +42,54 @@ struct ml_dsa_gen_ctx {
     size_t entropy_len;
 };
 
+#ifdef FIPS_MODULE
+static int ml_dsa_pairwise_test(const ML_DSA_KEY *key)
+{
+    OSSL_SELF_TEST *st = NULL;
+    OSSL_CALLBACK *cb = NULL;
+    OSSL_LIB_CTX *ctx;
+    void *cbarg = NULL;
+    static const uint8_t msg[] = { 80, 108, 117, 103, 104 };
+    uint8_t rnd[ML_DSA_ENTROPY_LEN];
+    uint8_t sig[ML_DSA_87_SIG_LEN];
```

Comment:
Should we dynamically allocate this, it is large (I do this elsewhere inside ml_dsa code)?

Reply:
I'm not seeing a real need.  It's 4896 bytes which isn't that large and this is restricted to the FIPS provider which implies a machine that supports shared loadable modules (i.e. has a MMU).

If it were non-FIPS code, I'd agree.